engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -973,48 +973,38 @@ struct devouring_plague_dot_t final : public priest_spell_t
   //   return dot_duration * ( tick_time( s ) / base_tick_time );
   // }
 
-  void append_damage( player_t* target )
+  void trigger_devouring_plague_dot( action_state_t* s )
   {
-    dot_t* dot = get_dot( target );
+    // Get existing DoT on the target
+    dot_t* dot = get_dot( s->target );
     assert( dot && dot->state && "Cannot append damage without active dot" );
-    assert( dot->current_action && "Cannot append damage to dot without action" );
-    // get remaining full ticks left (not partials, these will be rolled over)
-    // calculate total damage
-    timespan_t remaining_dot = dot->remains();                                       // 2.5s
-    remaining_dot            = remaining_dot - dot->time_to_next_tick();             // 2.5 - .5s = 2s
-    int num_full_ticks = as<int>( std::ceil( remaining_dot / dot->time_to_tick ) );  // 2s / ~1.5 = 1.3 = 1 full tick
-    auto calculated_tick_amount = dot->current_action->calculate_tick_amount( dot->state, 1.0 );
-    double total_damage         = num_full_ticks * calculated_tick_amount;
-    sim->print_debug( "{} {} old DP dot num_ticks={} tick_amount={} total_full_tick_damage={}", *player, *this,
-                      num_full_ticks, calculated_tick_amount, total_damage );
-    double total_ticks = dot->duration() / dot->time_to_tick;
-    if ( dot->time_to_next_tick() > timespan_t::from_seconds( 0 ) )
-    {
-      total_ticks += 1;
-    }
 
-    auto dot_state = debug_cast<devouring_plague_dot_state_t*>( dot->state );
+    // Get the existing duration before triggering the new DoT
+    timespan_t existing_duration = dot->remains();
+    priest_spell_t::trigger_dot( s );
 
-    double partial_tick_coeff = ( remaining_dot - ( num_full_ticks * dot->time_to_tick ) ) /
-                                dot->time_to_tick;  // 2s - ( 1 * 1.5 ) = .5 / 1.5 =~ 0.33
+    // Calculate the modifier for the future ticks
+    auto dot_state           = debug_cast<devouring_plague_dot_state_t*>( dot->state );
+    double existing_modifier = dot_state->rolling_dp_modifier;
+    dot_state->rolling_dp_modifier =
+        ( existing_duration * existing_modifier + composite_dot_duration( s ) ) / dot->remains();
```

Comment:
This line right now isn't removing the next tick or the partial tick, which was calculated before. Probably need to add it back. 

A good test for this is to do DP > GCD > GCD > DP, which should lead just to more ticks, but no damage increase.